### PR TITLE
RW-51-Get-task

### PR DESCRIPTION
**Pull request by lewis150193**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/RWA-51

### Change description ###
Get Task endpoint by ID

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ x ] No
```

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -233,6 +233,7 @@ dependencies {
 
   compile group: 'org.apache.tomcat.embed', name: 'tomcat-embed-core', version: '9.0.37'
   compile group: 'org.apache.tomcat.embed', name: 'tomcat-embed-websocket', version: '9.0.37'
+  compile group: 'commons-httpclient', name: 'commons-httpclient', version: '3.1'
```
Comment (iguannago) [inline build.gradle:236]: ummm, See here, please: https://tools.hmcts.net/confluence/display/WA/WA+Dev+Team+Guide

[We prefer to use Feign Client for Rest calls but will also use restTemplate when appropriate.]

Also, this library you use is tagged as "end of life" see here: https://hc.apache.org/httpclient-3.x/ 

I don't think it's a good idea to use it. 

You could just use the new apache common library https://hc.apache.org/httpcomponents-client-ga/, or restTemplate which is part of spring-boot or feign where we have examples in other projects or just the new java 11 HTTPClient: 
http://openjdk.java.net/groups/net/httpclient/intro.html

File: src/main/java/uk/gov/hmcts/reform/waworkflowapi/models/Task.java [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+    private String caseInstanceId;
+    private String caseDefinitionId;
+    private boolean suspended;
+    private String formKey;
+    private String tenantId;
+
+    public Task() {
+    }
+
+    public String getId() {
```
Comment (iguannago) [inline src/main/java/uk/gov/hmcts/reform/waworkflowapi/models/Task.java:31]: you can get rid of this boilter plate code by just adding the lombok annotation @Getter see here https://www.baeldung.com/intro-to-project-lombok

**Review by iguannago — APPROVED** [review]

File: src/integrationTest/java/uk/gov/hmcts/reform/waworkflowapi/controllers/GetCamundaTaskByIdTest.java [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+    @Test
+    void taskByIdTest() throws Exception {
+        when(getTaskController.getTask("025c59e3-dbe2-11ea-81e2-661816095024"))
+            .thenReturn("TestResponse");
+        mockMvc.perform(MockMvcRequestBuilders
+                            .get("/task/025c59e3-dbe2-11ea-81e2-661816095024")
+                            .accept(MediaType.APPLICATION_JSON))
+            .andExpect(status().isOk());
+    }
```
Comment (BenNzewi) [inline src/integrationTest/java/uk/gov/hmcts/reform/waworkflowapi/controllers/GetCamundaTaskByIdTest.java:36]: Please add edge case for failure

File: lombok.config [hunk 4]
```diff
@@ -0,0 +1,3 @@
+config.stopbubbling = true
+lombok.addLombokGeneratedAnnotation = true
+lombok.anyConstructor.addConstructorProperties = true
```
Comment (chrisgrimble) [inline lombok.config:3]: Why have we added lombok it has not been used. Also we decided not to use lombok on this project.

File: build.gradle [hunk 5]
```diff
@@ -252,6 +252,7 @@ dependencies {
 
   compile group: 'org.apache.tomcat.embed', name: 'tomcat-embed-core', version: '9.0.37'
   compile group: 'org.apache.tomcat.embed', name: 'tomcat-embed-websocket', version: '9.0.37'
+  compile group: 'org.projectlombok', name: 'lombok', version: '1.18.10'
```
Comment (chrisgrimble) [inline build.gradle:255]: Why have we added lombok it has not been used. Also we decided not to use lombok on this project.

File: src/main/java/uk/gov/hmcts/reform/waworkflowapi/controllers/startworkflow/GetTaskController.java [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+    public GetTaskController(CamundaTaskService camundaTaskService) {
+        this.camundaTaskService = camundaTaskService;
+    }
+
+    @GetMapping(path = "/task/{task-id}", produces = { MediaType.APPLICATION_JSON_VALUE })
+    public String getTask(@PathVariable("task-id") String id)  {
+        return camundaTaskService.getTask(id);
+    }
+
+}
```
Comment (chrisgrimble) [inline src/main/java/uk/gov/hmcts/reform/waworkflowapi/controllers/startworkflow/GetTaskController.java:26]: This endpoint should be part of task-management not workflow-api. Guess this whole change should have been done on that project not on here.

**Pull request merged**